### PR TITLE
Add boto datapipeline execution module and state

### PR DESCRIPTION
**Pull request by zackhsi**
The execution module provides basic CRUD operations.

The salt stack ensure that a pipeline with the matching attributes exists. A major use case is backing up dynamodb tables.

### CODE REVIEW

**Comment by cachedout** [comment]
Hi @zackhsi Thanks for this! There are a number of lint errors that need to be resovled before we can merge this. Please take a look and let me know if you have any questions. Once you've pushed a fix for those errors, we can take a look and hopefully get this merged. Thanks!

File: salt/modules/boto_datapipeline.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+'''
+Connection module for Amazon Data Pipeline
+
```
Comment (jfindlay) [inline salt/modules/boto_datapipeline.py]: @zackhsi, `.. versionadded:: Boron`.

File: salt/states/boto_datapipeline.py [hunk 2]
```diff
@@ -0,0 +1,575 @@
+# -*- coding: utf-8 -*-
+'''
+Manage Data Pipelines
+
```
Comment (jfindlay) [inline salt/states/boto_datapipeline.py]: @zackhsi, `.. versionadded:: Boron`.

**Comment by zackhsi** [comment]
Yup, will update PR soon with fixes.

Thanks for the quick comments :)

**Comment by jfindlay** [comment]
@zackhsi, one more lint:
```
salt/states/boto_datapipeline.py:55: [W1699(incompatible-py3-code), ] Incompatible Python 3 code found. Proposed fix:
23:00:06  import datetime
23:00:06  import difflib
23:00:06  import json
23:00:06 +from salt.ext.six.moves import zip
```

**Comment by zackhsi** [comment]
Thanks @jfindlay, I was just looking that that. I'll push that change now.

I'm not clear on root cause -- why do we need to import zip?

**Comment by jfindlay** [comment]
It is probably because zip behaves differently between python 2 and 3.

**Comment by zackhsi** [comment]
@jfindlay Failed the test SSHKnownHostsStateTest. How do I restart test?

```sh
23:44:17 FAIL [9.770s]: test_present (integration.states.ssh.SSHKnownHostsStateTest)
23:44:17 [CPU:21.4%|MEM:96.0%|Z:0] 
23:44:17 ----------------------------------------------------------------------
23:44:17 Traceback (most recent call last):
23:44:17   File "/testing/tests/integration/states/ssh.py", line 84, in test_present
23:44:17     ret, GITHUB_FINGERPRINT, keys=('new', 'fingerprint')
23:44:17   File "/testing/tests/integration/__init__.py", line 1534, in assertSaltStateChangesEqual
23:44:17     self.__getWithinSaltReturn(ret, keys), comparison
23:44:17   File "/testing/tests/integration/__init__.py", line 1438, in __getWithinSaltReturn
23:44:17     ''.join(['[\'{0}\']'.format(k) for k in keys]), part
23:44:17 AssertionError: Could not get ret['changes']['new']['fingerprint'] from salt's return: {'comment': 'Unable to receive remote host key', 'name': '192.30.252.129', 'start_time': '23:39:47.319421', '__id__': '192.30.252.129', 'duration': 5065.153, '__run_num__': 0, 'changes': {}, 'result': False}
```

**Comment by jfindlay** [comment]
@zackhsi, I'm not sure why that is failing, but it is unrelated.

**Comment by cachedout** [comment]
Looks great! Thank you, @zackhsi

**Comment by zackhsi** [comment]
Thanks Mike :palm_tree:

**Comment by cachedout** [comment]
My pleasure. Thanks again.

**Pull request merged**